dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/Acyclic.java

```diff
@@ -293,20 +294,20 @@ private BooleanFormula consistentSAT(Set<Tuple> toBeEncoded, EncodingContext con
             Set<Tuple> out = vertEleOutEdges.get(varOrderings.get(i));
             for (Tuple t : out) {
                 if (varOrderings.indexOf(t.getSecond()) > i && vertEleInEdges.get(t.getSecond()).contains(t)) {
-                    BooleanFormula cond = minSet.contains(t) ? bmgr.makeTrue() : getSMTCycleVar(t, fmgr);
-                    enc = bmgr.and(enc, bmgr.implication(cond, bmgr.not(getSMTCycleVar(t.getInverse(), fmgr))));
+                    BooleanFormula cond = minSet.contains(t) ? bmgr.makeTrue() : getSMTCycleVar(t, fmgr, bmgr);
+                    enc = bmgr.and(enc, bmgr.implication(cond, bmgr.not(getSMTCycleVar(t.getInverse(), fmgr, bmgr))));
                 }
             }
         }
 
         return enc;
     }
 
-    private BooleanFormula cycleVar(Event event, FormulaManager m) {
-        return m.getBooleanFormulaManager().makeVariable(String.format("cycle %s %d", m.escape(getName()), event.getCId()));
+    private BooleanFormula cycleVar(Event event, FormulaManager fmgr, BooleanFormulaManager bmgr) {
+        return bmgr.makeVariable("cycle_" + fmgr.escape(getName()) + "_" + event.repr());
     }
 
-    private BooleanFormula getSMTCycleVar(Tuple edge, FormulaManager m) {
-        return m.getBooleanFormulaManager().makeVariable(String.format("cycle %s %d %d", m.escape(getName()), edge.getFirst().getCId(), edge.getSecond().getCId()));
+    private BooleanFormula getSMTCycleVar(Tuple edge, FormulaManager fmgr, BooleanFormulaManager bmgr) {
+        return bmgr.makeVariable("cycle_" + fmgr.escape(getName()) + "_" + edge.getFirst().repr() + "_" + edge.getSecond().repr());
```

Comment:
I don't think passing a second argument was a good idea. The intention of my comment was "why do we need `fmgr` and not pass JUST `bmgr`?". Since we need `fmgr` for the `escape` and we can get `bmgr` from `fmgr`, I don't see the point in the second argument

Reply:
Then I will revert this.